<?xml version="1.0"?>
<launch>
	<arg name="global_ns" default="crs"/>	
	
	<let name="ns" value="$(var global_ns)/preview"/>
	<let name="robot_description_file" value="$(find-pkg-share crs_support)/urdf/crs.urdf"/>
	
	<node pkg="robot_state_publisher" exec="robot_state_publisher" name="$(var ns)/robot_state_publisher" namespace="$(var ns)"
		args="$(var robot_description_file)"/>
		
	<node pkg="crs_utils_py" exec="joint_state_publisher" name="$(var ns)/joint_state_publisher" namespace="$(var ns)" 
		output="screen">
		<param name="joint_names" 
			value="elbow_joint, shoulder_lift_joint, shoulder_pan_joint, wrist_1_joint, wrist_2_joint, wrist_3_joint"
			value-sep=", "/>
		<param name="joint_init_vals" value="'0.0', '0.0', '0.0', '0.0', '0.0', '0.0'" value-sep=", " />
		<param name="src_topics" value="input_joints" value-sep=", "/>
	</node>
	
	<!-- rviz -->
	<let name="rviz_config" value="$(find-pkg-share crs_application)/resources/crs.rviz"/>
	<node pkg="rviz2" exec="rviz2" args="-d $(var rviz_config)" output="log"/>
</launch>